src/runtime_src/core/pcie/driver/linux/xocl/subdev/hwmon_sdm.c

```diff
@@ -361,20 +361,28 @@ static int show_sensors_raw(struct xocl_hwmon_sdm *sdm, char *buf,
 {
 	xdev_handle_t xdev = xocl_get_xdev(sdm->pdev);
 	int ret = 0, kind;
-	size_t resp_len = RESP_LEN;
+	size_t resp_len = 0;
 	char* sdr_buf;
 	int repo_type;
 	uint64_t data_args = 0;
 
 	mutex_lock(&sdm->sdm_lock);
+	/* Collect the SDR size from VMR */
+	xocl_xgq_collect_sensors_get_size(xdev, &resp_len, repo_id);
```

Comment:
Hi @karthdmg-xilinx ,
IMO, call the function xocl_xgq_collect_sensors_get_size() only once in hwmon_sdm probe() and save the resp_size data in the struct xocl_hwmon_sdm for future usage in the driver.

Reply:
during hot reset (xclmgmt_hot_reset()), I believe re-probe wont happen so for this we may be able to read the new size information from VMR. So I added code for every request. overall sensor size may not change though even after hot reset, I'm not sure if it works correctly under various cases. correct me if I'm wrong.

Reply:
I think either approach is fine: pulling size every time or cache size. But please consider the corner case like if the cached size could be over flow the allocated memory buffer. I do not check the detail logics, but this case should be avoided in any combination.

Performance should be fine. Please test "xbutil examine" "xbutil query". I recall there was a CR for the xbutil runs over 1 second.

Reply:
buffer length is read per repo_id, we need array sensor_size params  to store in struct xocl_hwmon_sdm. if we consolidate all the repo_ids, then I see it is crossing 4k size

Reply:
yes, you need to define array of sensor_size[] and update it in probe(), reset() paths

Reply:
secondly, if not privileged (sdm->privileged), is it okay to send xgq command to calculate size? or the get size request should go through hwmon_sdm_read_from_peer() ?